### PR TITLE
JS: Type tracking through captured vars

### PR DESCRIPTION
**Pull request by asger-semmle**
As discussed in the meeting, this adds a bit of extra flow to type tracking.

The PR also contains a simplified version of the call graph metric, which no longer tries to guess whether a call is external. That was originally meant as a best-effort estimate of the call graph quality in  *absolute terms*, but it made it terrible for *relative* comparisons since classification of external calls could change between the two compared revisions, generating too much noise. Now it's much more stable, although in absolute terms we (still) don't know how how close we are to the ground truth.

Evaluations:
- [Metrics](https://git.semmle.com/asger/dist-compare-reports/tree/type-tracking-through-captured-vars_1568321319442) for call graph and number of DOM values show a modest improvement.
- [Performance](https://git.semmle.com/asger/dist-compare-reports/tree/type-tracking-through-captured-vars_1568456636994) and [re-run of the outlier](https://git.semmle.com/asger/dist-compare-reports/tree/type-tracking-through-captured-vars_1568724204097) shows that there is a minor cost.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/dataflow/Nodes.qll [hunk 1]
```diff
@@ -706,7 +706,8 @@ class ClassNode extends DataFlow::SourceNode {
     result = getAClassReference(t.continue()).getAnInstantiation()
     or
     t.start() and
-    result.(AnalyzedNode).getAValue() = getAbstractInstanceValue()
+    result.(AnalyzedNode).getAValue() = getAbstractInstanceValue() and
+    not result = any(DataFlow::ClassNode cls).getAReceiverNode()
```
Comment (esben-semmle) [inline javascript/ql/src/semmle/javascript/dataflow/Nodes.qll]: What is the motivation for this change?

Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/dataflow/Nodes.qll]: While investigating the call graph changes I noticed some of the new edges were spurious. This was the root cause but didn't surface without the new steps.
